src/praisonai-agents/praisonaiagents/memory/adapters/perseus_vault_adapter.py

```
@@ -0,0 +1,282 @@
+        cat = self._st_cat if memory_type == "short" else self._lt_cat
+        # Try both categories if type unspecified.
+        cats = [cat] if memory_type else [self._st_cat, self._lt_cat]
+        ok = False
+        for c in cats:
+            try:
+                self._client.call_tool("perseus_vault_forget", {
+                    "category": c, "key": memory_id, "reason": "PraisonAI delete_memory",
+                })
+                ok = True
+            except Exception:
+                continue
```

Ruff (0.15.20)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)